### ISSUE TITLE
SSH Server only offering one kex algo, config options aren't affecting this.

### ISSUE THREAD

**Issue by ChristopherJMiller** [ContainerSSH/ContainerSSH#535]
## Describe the bug

It appears that during the ssh handshake, ContainerSSH is only offering one type of key exchange algorithm, `ssh-rsa`. Adding additional key exchange algorithms to the config file appear to not do anything.

In these examples, using a key generated with the following command:

`ssh-keygen -t rsa -b 4096 -N ""`

Output with no `kex` config option specificed:
```
{"timestamp":"2022-10-28T20:29:29Z","level":"debug","code":"SSH_CONNECTED","message":"Client connected","details":{"connectionId":"d4897a3561d2468eba0e67d8c7e17264","module":"ssh","remoteAddr":"::1"}}
{"timestamp":"2022-10-28T20:29:29Z","level":"info","code":"SSH_HANDSHAKE_FAILED","message":"SSH handshake failed (ssh: no common algorithm for host key; client offered: [rsa-sha2-512-cert-v01@openssh.com rsa-sha2-256-cert-v01@openssh.com rsa-sha2-512 rsa-sha2-256 ssh-ed25519-cert-v01@openssh.com ecdsa-sha2-nistp256-cert-v01@openssh.com ecdsa-sha2-nistp384-cert-v01@openssh.com ecdsa-sha2-nistp521-cert-v01@openssh.com ssh-ed25519 ecdsa-sha2-nistp256 ecdsa-sha2-nistp384 ecdsa-sha2-nistp521], server offered: [ssh-rsa])","details":{"connectionId":"d4897a3561d2468eba0e67d8c7e17264","module":"ssh","remoteAddr":"::1"}}
{"timestamp":"2022-10-28T20:29:29Z","level":"debug","code":"SSH_DISCONNECTED","message":"Client disconnected","details":{"connectionId":"d4897a3561d2468eba0e67d8c7e17264","module":"ssh","remoteAddr":"::1"}}
```

If I modify the config file to include the kex configuration (in this case, choosing an algo at random from the docs list):
```
    ssh:
      kex:
        - "ecdh-sha2-nistp521"
    ...
```

Containerssh is restarted, and we get the same output, no new algo offered:

```
{"timestamp":"2022-10-28T20:37:15Z","level":"debug","code":"SSH_CONNECTED","message":"Client connected","details":{"connectionId":"cfd83bda7fc1412db9d90040cb0870ed","module":"ssh","remoteAddr":"::1"}}
{"timestamp":"2022-10-28T20:37:15Z","level":"info","code":"SSH_HANDSHAKE_FAILED","message":"SSH handshake failed (ssh: no common algorithm for host key; client offered: [rsa-sha2-512-cert-v01@openssh.com rsa-sha2-256-cert-v01@openssh.com rsa-sha2-512 rsa-sha2-256 ssh-ed25519-cert-v01@openssh.com ecdsa-sha2-nistp256-cert-v01@openssh.com ecdsa-sha2-nistp384-cert-v01@openssh.com ecdsa-sha2-nistp521-cert-v01@openssh.com ssh-ed25519 ecdsa-sha2-nistp256 ecdsa-sha2-nistp384 ecdsa-sha2-nistp521], server offered: [ssh-rsa])","details":{"connectionId":"cfd83bda7fc1412db9d90040cb0870ed","module":"ssh","remoteAddr":"::1"}}
{"timestamp":"2022-10-28T20:37:15Z","level":"debug","code":"SSH_DISCONNECTED","message":"Client disconnected","details":{"connectionId":"cfd83bda7fc1412db9d90040cb0870ed","module":"ssh","remoteAddr":"::1"}}
```

SSH Client Version: OpenSSH_9.0p1, LibreSSL 3.3.6 on MacOS Ventura

## To Reproduce

1. Run containerssh using the 0.4.1 tagged container
2. Generate a RSA 4096bit keypair
3. Use the default config example from the docs
4. Perform a basic ssh login

## Expected behavior

Successful login

## Version

0.4.1

## Additional Context

- This deployment is configured with auth and config endpoints
- This is deployed with a kubernetes backend, following docs recommendations for storing the config file in a config map and then mounting that readonly
- The entire deployment namespace was brought down and re-deployed between tests to ensure everything is applied correctly.

**Comment by anderwmo** [comment]
Hi,
I am in the same situation as @jmcclelland and was wondering if there is any updates on this issue?  It's only three months since the last update on the situatuon, but i thought it was worth a shot to ask.

**Comment by tsipinakis** [comment]
Closing as fixed with 0.5.
